tools/gen-bootstrap.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+# The script is designed for simplification of new bootstrap.snap generation.
+# Just run it in tarantool build directory after updating update.lua (which
+# introduces new data version) and examine its output.
+# The script will try to generate bootstrap.snap in the current directory, and
+# in case of success will offer to copy the file to the source tree.
+# The manual instruction by which the script was created is available here:
+MANUAL_LINK="https://github.com/tarantool/tarantool/wiki/How-to-generate-new-bootstrap-snapshot"
+# Now the script does the following:
+ACTIONS=( "rebuild tarantool.")
+ACTIONS+=("run box.cfg{} box.internal.bootstrap() by tarantool in a clean dir.")
+ACTIONS+=("take the latest snapshot file an rename it to bootstrap.snap file.")
+ACTIONS+=("edit the file and change line with VClock to an empty set: {}.")
+
+TNT_BINARY=./src/tarantool
+TEMP_DIR=./gen-bootstrap-tmp
+
+if ! "$TNT_BINARY" --version >/dev/null 2>&1; then
+    echo "Must be run from tnt build dir, with tarantool binary built in it."
+    exit 1
+fi
+
+if ! xxd -v >/dev/null 2>&1; then
+    echo "xxd must be installed."
+    exit 1
+fi
+
+if [[ -f "$TEMP_DIR" ]]; then
+    echo "Regular file $TEMP_DIR was found, has no idea what to do, exiting."
+    exit 1
+fi
+
+if [[ -d "$TEMP_DIR" ]]; then
+    echo "Folder $TEMP_DIR was found, it seems that previous run was failed."
+    read -p "Drop the folder and proceed [Yn]?" yn
+    case ${yn:-y} in
+        [Yy]* ) rm -r "$TEMP_DIR";;
+        [Nn]* ) echo "OK, exiting."; exit 0;;
+        * ) echo "Wrong answer."; exit 1;;
+    esac
+fi
+
+if ! cmake --build . -j; then
+    echo "Failed to rebuild tarantool."
+    exit 1
+fi
+
+mkdir "$TEMP_DIR"
+cd "$TEMP_DIR"
+
+if ! ../"$TNT_BINARY" -e "box.cfg{} box.internal.bootstrap() os.exit(0)"; then
+    echo "Failed to run script by tarantool, exiting."
+    exit 1
+fi
+
+mv $(ls *.snap | sort | tail -n 1) ./bootstrap.snap
+
+# Clear VClock field of the file.
+# Actually all the code below does only simple command:
+# sed -i "s/VClock: {.*}/VClock: {}/" ./bootstrap.snap
```

Comment:
The code below looks complex. Wouldn't it be better to use `sed` to replace the first match instead?

```
sed -i "s/VClock: {.*}/VClock: {}/"
```

Reply:
You copy-pasted my line, but we discussed offline that there was special option for set..

Reply:
Sorry, I meant to paste this:

```
sed -i '0,/^VClock: {}$/{s//VClock: foo/}'
```

Reply:
Thanks, I added you command to comment, but I don't want to drop hex/binary code because it was a challenge after which I decided to write the script.
So let's leave it as is and iff the code will need to be supported we can simply switch to sed version.